### PR TITLE
Update to Angular 9

### PR DESCRIPTION
**Pull request by tobiasschweizer**
Update to Angular 9.

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -75,31 +82,34 @@ export class AppComponent {
 @Component({
   selector: 'app-calendar-header',
   template: `
-    <mat-select placeholder="Calendar Format" [formControl]="form.controls['calendar']">
-      <mat-option *ngFor="let cal of supportedCalendars" [value]="cal">{{cal}}</mat-option>
-    </mat-select>
+        <mat-select placeholder="Calendar" [formControl]="calendar">
```
Comment (tobiasschweizer) [inline src/app/app.component.ts:85]: @flavens I tried to wrap the `mat-select` with `mat-form-field` but this led to a CSS problem. In `styles.scss` I added `.mat-form-field-infix` but it did not have an effect. Could you give me a hand here?

Comment (flavens) [reply src/app/app.component.ts:85]: @tobiasschweizer yes I can this afternoon

Comment (tobiasschweizer) [reply src/app/app.component.ts:85]: > @tobiasschweizer yes I can this afternoon

great, thx

Comment (flavens) [reply src/app/app.component.ts:85]: @tobiasschweizer how can I help? what do you want to do? how can I run it on my machine?

Comment (tobiasschweizer) [reply src/app/app.component.ts:85]: I tried to put the whole `mat-select` in a `mat-form-field` 

```html
    <mat-form-field>
        <mat-select placeholder="Calendar" [formControl]="calendar">
          <mat-option *ngFor="let cal of supportedCalendars" [value]="cal">{{cal}}</mat-option>
        </mat-select>
    </mat-form-field>
    <mat-calendar-header></mat-calendar-header>
```

and then the CSS seemed broken. I suppose the styles in `styles.scss`weren't applied any more.

Steps to run this locally:
- check out this branch
- run `npm run build-lib`
- run `npm run ng s`
- access `http://localhost:4200/` and open one of the datepickers

Screenshot 2020-04-20 at 16 49 05

I tried to add `.mat-form-field-infix` to `styles.scss` but it didn't have any effect.

Comment (tobiasschweizer) [reply src/app/app.component.ts:85]: I think I just don't understand how scss works ...

Comment (flavens) [reply src/app/app.component.ts:85]: where did you add this code block?
```
<mat-form-field>
        <mat-select placeholder="Calendar" [formControl]="calendar">
          <mat-option *ngFor="let cal of supportedCalendars" [value]="cal">{{cal}}</mat-option>
        </mat-select>
    </mat-form-field>
    <mat-calendar-header></mat-calendar-header>
```

Comment (tobiasschweizer) [reply src/app/app.component.ts:85]:
```css
app-calendar-header {
  .mat-select {
    background: #00796b;
    color: white;
    padding: 20px;
    margin-bottom: -20px;
    box-sizing: border-box;
  }

  .mat-select-value,
  .mat-select-arrow {
    color: #fff;
  }
}

.mat-datepicker-content {
  height: 400px;
}
```

I tried to add `.mat-form-field-infix` as a parent selector

File: src/app/app.component.ts [hunk 2]
```diff
@@ -75,31 +82,34 @@ export class AppComponent {
...
+          <mat-option *ngFor="let cal of supportedCalendars" [value]="cal">{{cal}}</mat-option>
+        </mat-select>
     <mat-calendar-header></mat-calendar-header>
   `,
   styleUrls: []
 })
 export class HeaderComponent<D> implements OnInit {
-  constructor(@Host() private _calendar: MatCalendar<JDNConvertibleCalendar>,
+  constructor(private _calendar: MatCalendar<JDNConvertibleCalendar>,
```
Comment (tobiasschweizer) [inline src/app/app.component.ts:93]: @flavens We will have to remove `@Host` in https://github.com/dasch-swiss/knora-ui-ng-lib/blob/714826ad197eea8dd16286510e65c39b1d1a8aec/projects/knora-ui/src/lib/viewer/values/date-value/calendar-header/calendar-header.component.ts#L16

Comment (flavens) [reply src/app/app.component.ts:93]: ok

Comment (tobiasschweizer) [reply src/app/app.component.ts:93]: I got it from here: https://github.com/angular/angular/issues/31539#issuecomment-511290841

File: src/app/app.module.ts [hunk 3]
```diff
@@ -30,10 +30,7 @@ import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
     BrowserAnimationsModule
   ],
   providers: [],
-  bootstrap: [AppComponent],
-  entryComponents: [
```
Comment (tobiasschweizer) [inline src/app/app.module.ts:34]: @flavens We can then safely remove `entryComponents` from https://github.com/dasch-swiss/knora-ui-ng-lib/blob/714826ad197eea8dd16286510e65c39b1d1a8aec/projects/knora-ui/src/lib/viewer/viewer.module.ts#L80

File: package.json [hunk 4]
```diff
@@ -7,51 +7,51 @@
     "build": "ng build",
     "test": "ng test",
     "test-lib": "ng t --watch=false --browsers=ChromeHeadless --project jdnconvertiblecalendardateadapter",
-    "build-lib": "ng b --project jdnconvertiblecalendardateadapter",
+    "build-lib": "ng b --prod --project jdnconvertiblecalendardateadapter",
```
Comment (tobiasschweizer) [inline package.json:10]: https://angular.io/guide/creating-libraries#publishing-your-library Ivy should not be used when building a library. 

Actually, the compiler shows a warning when not using the `--prod` flag:

> Building Angular Package
> ******************************************************************************
> It is not recommended to publish Ivy libraries to NPM repositories.
> Read more here: https://v9.angular.io/guide/ivy#maintaining-library-compatibility
> ******************************************************************************

**Review by flavens — APPROVED** [review]

**Comment by tobiasschweizer** [comment]
@flavens Thanks for the review!

**Pull request merged**